IdentityCore/src/cache/crypto/MSIDSymmetricKey.m

```diff
@@ -177,11 +209,11 @@ - (uint8_t *)kdfCounterMode:(uint8_t *)keyDerivationKey
 /*
  *Function used to shift data by 4 byte and insert ctr in the first 4 bytes.
  */
-- (uint8_t *)updateDataInput:(uint32_t)ctr
+- (uint8_t *)updateDataInput:(uint8_t)ctr
```

Comment:
Since counter variable is initialized as a uint32_t, let's keep it as uint32_t. 
I know that theoretically, L/h = 4096/256 = 16 = 0x00000010, so it probably won't use the first 24 bits, but I think it would be good to keep it uint32_t.

Reply:
Yes it's true. My plan is to revert back to the original code and I will refactor it to support different input for other spec (e.g different label string...). unint32_t looks better, you're right let keep that.
sizeof(uint32_t) == 4 bytes so I think either way is fine. I personal prefer 4 because that tell me the size will be input + 4 bytes. In other hand if it said `sizeof(uint32_t)` I need extra step to find out what's the exact value :D